internal/controllers/image/upload.go

```diff
@@ -129,18 +128,21 @@ func (r *orcImageReconciler) uploadImageContent(ctx context.Context, orcImage *o
 		}
 	}
 
-	err = r.updateStatus(ctx, orcImage, withResource(glanceImage),
-		withIncrementDownloadAttempts(),
-		withProgressMessage(downloadingMessage("Starting image upload", orcImage)))
+	err = setDownloadingStatus(ctx, true, "Starting image upload", orcv1alpha1.ConditionReasonProgressing, metav1.ConditionTrue, orcImage, actuator.k8sClient)
 	if err != nil {
 		return err
 	}
 
-	err = imageClient.UploadData(ctx, glanceImage.ID, reader)
+	err = actuator.osClient.UploadData(ctx, glanceImage.ID, reader)
 	if err != nil {
+		reason := orcv1alpha1.ConditionReasonTransientError
 		if orcerrors.IsInvalidError(err) {
-			err = orcerrors.Terminal(orcv1alpha1.ConditionReasonInvalidConfiguration, err.Error(), err)
+			reason = orcv1alpha1.ConditionReasonInvalidConfiguration
+			err = orcerrors.Terminal(reason, err.Error(), err)
 		}
+		// We don't really care about catching errors when setting
+		// download status, we don't want to hide previous error
+		_ = setDownloadingStatus(ctx, false, err.Error(), reason, metav1.ConditionFalse, orcImage, actuator.k8sClient)
 		return fmt.Errorf("error writing data to glance: %w", err)
```

Comment:
`errors.Join()` ftw.

```suggestion
		return errors.Join(
		    setDownloadingStatus(ctx, false, err.Error(), reason, metav1.ConditionFalse, orcImage, actuator.k8sClient),
		    fmt.Errorf("error writing data to glance: %w", err),
		)
```